components/announcement_bar/contact_sales/contact_us.tsx

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+
+import {trackEvent} from 'actions/telemetry_actions';
+import './contact_us.scss';
+
+export interface Props {
+    buttonTextElement?: JSX.Element;
+    eventID?: string;
```

Comment:
My gut feeling says that these should be required and passed down instead of optional.
Do you think it's better to keep as a hybrid like component? I'm not a huge fan of the `|| contact-sales` aspect.

I'd be more in favor of making these required and passing them down from the contact sales so we don't need the `||` operator.

Or you could do it like the purchase link component where only the EventID is optional but buttomTextElement is required. This would be an ideal solution.

Reply:
Well, the whole idea of creating a contact us component in one of the comments was to have it with the default texts, and only as a flexibility, to allow overriding the content, if ever needed anywhere in the app. 

Making the params mandator defeats that purpose.

Removing the params vs having optional params doesn't make much difference as I don't think it's worth the effort going in and making these small changes that have no effect.